snuba/datasets/discover.py

```diff
@@ -327,13 +248,15 @@ def column_expr(
         parsing_context: ParsingContext,
         table_alias: str = "",
     ):
-        detected_dataset = detect_dataset(
+        detected_dataset = detect_table(
             query, self.__events_columns, self.__transactions_columns
         )
 
+        # Assignment in order not to loose track there is a type error in the Query class.
```

Comment:
What does this mean?

Reply:
I noticed this type error since vscode now shows them inline. The error is in the query class. It does not make sense get_granularity would return optional to me.
So instead of fixing it here by checking for None (which should not be needed), adding this assignment should make it clear that the method return value's type is wrong.
Clearly it was not so clear, so I will move the same check into the query class to isolate the type error there.

Reply:
This was more of a question about what the error was because there was no indication of wh the error was or how the change solved or mitigated it in the comment itself. This may have been a reasonable solution but it's impossible to tell without running mypy